exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/JoinStatus.java

```diff
@@ -101,8 +101,12 @@ public final void resetOutputPos() {
   }
 
   public final boolean isOutgoingBatchFull() {
-    Preconditions.checkArgument(outputPosition <= OUTPUT_BATCH_SIZE);
-    return outputPosition == OUTPUT_BATCH_SIZE;
+    Preconditions.checkArgument(outputPosition <= outputRowCount);
+    return outputPosition == outputRowCount;
```

Comment:
Maybe be just a bit more paranoid? `outputPosition >= outputRowCount`?

And, while we're at it, maybe `outputRowCount` -> `targetOutputRowCount`? To make clear that the value is our target, not the actual, current row count.